src/Microsoft.Data.Analysis/IDataView.Extension.cs

```diff
@@ -0,0 +1,128 @@
+            }
+
+            List<DataViewSchema.Column> activeColumns = new List<DataViewSchema.Column>();
+            foreach (DataViewSchema.Column column in schema)
+            {
+                long length = maxRows >= 0 ? maxRows : long.MaxValue;
+                length = Math.Min(length, dataView.GetRowCount() ?? 0);
+                if (column.IsHidden || (selectColumnsSet != null && !selectColumnsSet.Contains(column.Name)))
+                {
+                    continue;
+                }
+
+                activeColumns.Add(column);
+                DataViewType type = column.Type;
+                if (type == BooleanDataViewType.Instance)
+                {
+                    columns.Add(new BooleanDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Byte)
+                {
+                    columns.Add(new ByteDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Double)
+                {
+                    columns.Add(new DoubleDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Single)
+                {
+                    columns.Add(new SingleDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Int32)
+                {
+                    columns.Add(new Int32DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Int64)
+                {
+                    columns.Add(new Int64DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.SByte)
+                {
+                    columns.Add(new SByteDataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.Int16)
+                {
+                    columns.Add(new Int16DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.UInt32)
+                {
+                    columns.Add(new UInt32DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.UInt64)
+                {
+                    columns.Add(new UInt64DataFrameColumn(column.Name, length));
+                }
+                else if (type == NumberDataViewType.UInt16)
+                {
+                    columns.Add(new UInt16DataFrameColumn(column.Name, length));
+                }
+                else if (type == TextDataViewType.Instance)
+                {
+                    columns.Add(new StringDataFrameColumn(column.Name, length));
+                }
+                else
+                {
+                    throw new NotSupportedException(String.Format(Microsoft.Data.Strings.NotSupportedColumnType, type.RawType.Name));
```

Comment:
This will cause a problem for vector types in IDataView I think. We'd need to add support for vector columns in DataFrame to support that scenario. I'll open a bug

Reply:
https://github.com/dotnet/machinelearning/issues/5721